Add maps for all colors, document how to extend color utilities

mdo opened the pull request:
The goal of this PR is to give folks the tools they need to extend our full color palette into any number of utilities, from `color` to `background-color` and beyond. I've added the maps to the `_variables.scss` file as our existing color variables all live there right now. I've also added some docs guidance for how to put it to use to the Customize > Color page. This also includes adding a new `map-merge-multiple()` function for combining several Sass maps, since otherwise we'd need to do nested maps for each merge.

/cc @twbs/css-review 

---

Closes #32236, closes #33412.

Preview: <https://deploy-preview-32319--twbs-bootstrap.netlify.app/docs/5.0/customize/color/#generating-utilities>

XhmikosR commented on site/content/docs/5.0/customize/color.md:
```diff
@@ -106,3 +106,40 @@ Here's how you can use these in your Sass:
+Bootstrap doesn't include `color` and `background-color` utilities for every individual color, but you can generate these yourself with our [utility API]({{< docsref "/utilities/api" >}}).
+
+1. To start, make sure you've imported our functions, variables, mixins, and utilities.
+2. Then include the `utilities/_color-extended.scss` stylesheet to access a full suite of color Sass maps.
```
Stylesheet does not seem right? Also a full suite or the full suite?

patrickhlauke replied:
i'd circumvent the first question by just using "Then include `utilities/_color-extend.scss` to access..."

and in this case, either `a` or `the` work, but i think i'd lean more towards `the`

mdo committed:
Add maps for all colors, document how to extend color utilities

mdo committed:
Updates to make this more functional for v5

- Moves color maps into _variables.scss for now (felt weird to split them out when the variables are there)
- Adds a new function, map-merge-multiple(), and updates docs that reference combo-map
- Updates code snippet and docs guidance about how to extend the color utils to handle this

ffoodd commented on site/content/docs/5.0/customize/color.md:
```diff
@@ -106,3 +106,45 @@ Here's how you can use these in your Sass:
+Bootstrap doesn't include `color` and `background-color` utilities for every color variable, but you can generate these yourself with our [utility API]({{< docsref "/utilities/api" >}}) and our extended Sass maps added in v5.1.0.
+
+1. To start, make sure you've imported our functions, variables, mixins, and utilities.
+2. Use our `combo-map()` function to quickly merge multiple Sass maps together in a new map.
```
Should be `map-merge-multiple()`, shouldn't it?
```suggestion
2. Use our `map-merge-multiple()` function to quickly merge multiple Sass maps together in a new map.
```

ffoodd reviewed (approved):
A function name to change in the docs, but that's great!

ffoodd committed:
Update site/content/docs/5.0/customize/color.md

XhmikosR committed:
Merge branch 'main' into extend-colors

mdo committed:
Merge branch 'main' into extend-colors
